### ISSUE TITLE
Installation issues

### ISSUE THREAD

**Issue by Dhiraj240** [publiclab/spectral-workbench.js#77]
```
sudo npm install 

> contextify@0.1.15 install /home/dhiraj/spectral-workbench.js/node_modules/contextify
> node-gyp rebuild

gyp ERR! configure error 
gyp ERR! stack Error: EACCES: permission denied, mkdir '/home/dhiraj/spectral-workbench.js/node_modules/contextify/build'
gyp ERR! System Linux 4.15.0-29-generic
gyp ERR! command "/usr/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/dhiraj/spectral-workbench.js/node_modules/contextify
gyp ERR! node -v v8.15.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
npm WARN grunt-contrib-watch@0.6.1 requires a peer of grunt@~0.4.0 but none is installed. You must install peer dependencies yourself.
npm WARN grunt-contrib-connect@0.8.0 requires a peer of grunt@~0.4.0 but none is installed. You must install peer dependencies yourself.
npm WARN grunt-contrib-jshint@0.10.0 requires a peer of grunt@~0.4.0 but none is installed. You must install peer dependencies yourself.
npm WARN grunt-contrib-uglify@0.5.1 requires a peer of grunt@~0.4.0 but none is installed. You must install peer dependencies yourself.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! contextify@0.1.15 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the contextify@0.1.15 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/dhiraj/.npm/_logs/2019-02-23T08_25_01_905Z-debug.log
```
Need help on installation. @jywarren  is it possible that you can guide via a gif or video.

**Comment by jywarren** [comment]
Hmm... i wonder if this is related to using sudo? Can you try without sudo?

**Comment by Dhiraj240** [comment]
i get the below one  using npm install
```
F:\spectral-workbench.js\node_modules\contextify>if not defined npm_config_node_
gyp (node "C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\n
ode-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node
"C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js"
 rebuild )
gyp ERR! configure error
gyp ERR! stack Error: Command failed: C:\Users\LENOVO\AppData\Local\Programs\Pyt
hon\Python36\python.EXE -c import sys; print "%s.%s.%s" % sys.version_info[:3];
gyp ERR! stack   File "<string>", line 1
gyp ERR! stack     import sys; print "%s.%s.%s" % sys.version_info[:3];
gyp ERR! stack                                ^
gyp ERR! stack SyntaxError: invalid syntax
gyp ERR! stack
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:294:12)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at maybeClose (internal/child_process.js:962:16)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_proces
s.js:251:5)
gyp ERR! System Windows_NT 6.3.9600
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodej
s\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd F:\spectral-workbench.js\node_modules\contextify
gyp ERR! node -v v10.15.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
npm WARN grunt-contrib-watch@0.6.1 requires a peer of grunt@~0.4.0 but none is i
nstalled. You must install peer dependencies yourself.
npm WARN grunt-contrib-uglify@0.5.1 requires a peer of grunt@~0.4.0 but none is
installed. You must install peer dependencies yourself.
npm WARN grunt-contrib-connect@0.8.0 requires a peer of grunt@~0.4.0 but none is
 installed. You must install peer dependencies yourself.
npm WARN grunt-contrib-jshint@0.10.0 requires a peer of grunt@~0.4.0 but none is
 installed. You must install peer dependencies yourself.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! contextify@0.1.15 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the contextify@0.1.15 install script.
npm ERR! This is probably not a problem with npm. There is likely additional log
ging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\LENOVO\AppData\Roaming\npm-cache\_logs\2019-03-04T10_16_33
_084Z-debug.log
```

**Comment by jywarren** [comment]
Oh, hmm, that seems like it may be a nodejs installation issue? Do you have
nodejs installed and running normally? If not, I can help with that!

**Comment by Dhiraj240** [comment]
It is installed and working.

**Comment by sidntrivedi012** [comment]
@Dhiraj240 I was facing this and solved it by `npm i grunt@~0.4.0`.

**Comment by sidntrivedi012** [comment]
The grunt error was resolved but the `node-gyp rebuild` error related to `contextify` still persists.
```js
> contextify@0.1.15 install /Users/sidntrivedi012/Code/spectral-workbench.js/node_modules/contextify
> node-gyp rebuild

  CXX(target) Release/obj.target/contextify/src/contextify.o
../src/contextify.cc:62:65: warning: 'ToObject' is deprecated [-Wdeprecated-declarations]
        ContextifyContext* ctx = new ContextifyContext(info[0]->ToObject());
                                                                ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8.h:2455:10: note: 'ToObject' has been explicitly
      marked deprecated here
  inline V8_DEPRECATED("Use maybe version", Local<Object> ToObject() const);
         ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8config.h:327:29: note: expanded from macro
      'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/contextify.cc:131:56: error: no matching member function for call to 'NewInstance'
        Local<Object> wrapper = Nan::New(constructor)->NewInstance();
                                ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8.h:3996:44: note: candidate function not viable:
      requires single argument 'context', but no arguments were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Object> NewInstance(
                                           ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8.h:3993:44: note: candidate function not viable:
      requires 3 arguments, but 0 were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Object> NewInstance(
                                           ^
../src/contextify.cc:150:16: error: no member named 'SetAccessCheckCallbacks' in 'v8::ObjectTemplate'
        otmpl->SetAccessCheckCallbacks(GlobalPropertyNamedAccessCheck,
        ~~~~~  ^
../src/contextify.cc:176:43: warning: 'ToObject' is deprecated [-Wdeprecated-declarations]
        Local<Object> data = info.Data()->ToObject();
                                          ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8.h:2455:10: note: 'ToObject' has been explicitly
      marked deprecated here
  inline V8_DEPRECATED("Use maybe version", Local<Object> ToObject() const);
         ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8config.h:327:29: note: expanded from macro
      'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/contextify.cc:182:80: error: too few arguments to function call, expected 2, have 1
        Local<Value> rv = Nan::New(ctx->sandbox)->GetRealNamedProperty(property);
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~         ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8.h:3562:3: note: 'GetRealNamedProperty' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Value> GetRealNamedProperty(
  ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8config.h:416:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/contextify.cc:192:43: warning: 'ToObject' is deprecated [-Wdeprecated-declarations]
        Local<Object> data = info.Data()->ToObject();
                                          ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8.h:2455:10: note: 'ToObject' has been explicitly
      marked deprecated here
  inline V8_DEPRECATED("Use maybe version", Local<Object> ToObject() const);
         ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8config.h:327:29: note: expanded from macro
      'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/contextify.cc:203:43: warning: 'ToObject' is deprecated [-Wdeprecated-declarations]
        Local<Object> data = info.Data()->ToObject();
                                          ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8.h:2455:10: note: 'ToObject' has been explicitly
      marked deprecated here
  inline V8_DEPRECATED("Use maybe version", Local<Object> ToObject() const);
         ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8config.h:327:29: note: expanded from macro
      'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/contextify.cc:209:67: error: too few arguments to function call, expected 2, have 1
        if (!Nan::New(ctx->sandbox)->GetRealNamedProperty(property).IsEmpty() ||
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~         ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8.h:3562:3: note: 'GetRealNamedProperty' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Value> GetRealNamedProperty(
  ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8config.h:416:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/contextify.cc:210:71: error: too few arguments to function call, expected 2, have 1
            !Nan::New(ctx->proxyGlobal)->GetRealNamedProperty(property).IsEmpty()) {
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~         ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8.h:3562:3: note: 'GetRealNamedProperty' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Value> GetRealNamedProperty(
  ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8config.h:416:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/contextify.cc:218:43: warning: 'ToObject' is deprecated [-Wdeprecated-declarations]
        Local<Object> data = info.Data()->ToObject();
                                          ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8.h:2455:10: note: 'ToObject' has been explicitly
      marked deprecated here
  inline V8_DEPRECATED("Use maybe version", Local<Object> ToObject() const);
         ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8config.h:327:29: note: expanded from macro
      'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/contextify.cc:229:43: warning: 'ToObject' is deprecated [-Wdeprecated-declarations]
        Local<Object> data = info.Data()->ToObject();
                                          ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8.h:2455:10: note: 'ToObject' has been explicitly
      marked deprecated here
  inline V8_DEPRECATED("Use maybe version", Local<Object> ToObject() const);
         ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8config.h:327:29: note: expanded from macro
      'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/contextify.cc:319:53: warning: 'ToObject' is deprecated [-Wdeprecated-declarations]
        if (!ContextifyContext::InstanceOf(info[0]->ToObject()))
                                                    ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8.h:2455:10: note: 'ToObject' has been explicitly
      marked deprecated here
  inline V8_DEPRECATED("Use maybe version", Local<Object> ToObject() const);
         ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8config.h:327:29: note: expanded from macro
      'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/contextify.cc:322:86: warning: 'ToObject' is deprecated [-Wdeprecated-declarations]
        ContextifyContext* ctx = Nan::ObjectWrap::Unwrap<ContextifyContext>(info[0]->ToObject());
                                                                                     ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8.h:2455:10: note: 'ToObject' has been explicitly
      marked deprecated here
  inline V8_DEPRECATED("Use maybe version", Local<Object> ToObject() const);
         ^
/Users/sidntrivedi012/Library/Caches/node-gyp/10.16.0/include/node/v8config.h:327:29: note: expanded from macro
      'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
8 warnings and 5 errors generated.
make: *** [Release/obj.target/contextify/src/contextify.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:190:23)
gyp ERR! stack     at ChildProcess.emit (events.js:198:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:248:12)
gyp ERR! System Darwin 18.5.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/sidntrivedi012/Code/spectral-workbench.js/node_modules/contextify
gyp ERR! node -v v10.16.0
gyp ERR! node-gyp -v v5.0.2
gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! contextify@0.1.15 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the contextify@0.1.15 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/sidntrivedi012/.npm/_logs/2019-07-21T05_13_50_684Z-debug.log
```

**Comment by sidntrivedi012** [comment]
Issue resolved by updating all the packages. Maybe one of the outdated packages would be using contextify as a package but, now it has been included in the Node core after v0.12 . Got this feedback from the issue - https://github.com/nodejs/node-gyp/issues/1837
